### PR TITLE
fix(Create/EditChatPopup): Added permissions related section

### LINKED ISSUE #12142: [Communities] Add permissions related content in create/edit channel popup
### Description
Figma:
https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2975-488707&mode=design&t=7bhUfNQmmUXSOz0g-4

pp

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #12142

Depends on: https://github.com/status-im/status-desktop/pull/12143
Depends on: https://github.com/status-im/status-desktop/issues/12609

### What does the PR do
Create/EditChatPopup: Added permissions related section

### Affected areas
Create/EditChatPopup

### Screenshot of functionality (including design for comparison)

https://github.com/status-im/status-desktop/assets/31625338/6aa94cf4-e791-4819-a8eb-c223a574b398

https://github.com/status-im/status-desktop/assets/31625338/f99d03a5-0ed2-432a-8c8f-b0dafffa1cda

https://github.com/status-im/status-desktop/assets/31625338/649e9b59-2ac8-4e0f-9756-1ff8e4293c45

### CODE REVIEW

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 1]
```diff
@@ -605,6 +639,136 @@ StatusStackModal {
...
+                                horizontalAlignment: Text.AlignHCenter
+                                verticalAlignment: Text.AlignVCenter
+                                text: qsTr("No channel permissions")
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        ColorPanel {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:723]: Why is this here as a separate page? You are basically reverting my fixes from https://github.com/status-im/status-desktop/pull/12143; do you have some old/unmerged copy of this popup?

**Review by caybro — DISMISSED** [review]

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 2]
```diff
@@ -605,6 +638,113 @@ StatusStackModal {
...
+                }
+                RowLayout {
+                    Layout.fillWidth: true
+                    Layout.preferredHeight: 56
+                    Layout.leftMargin: Style.current.padding
+                    Layout.rightMargin: Style.current.padding
+                    StatusBaseText {
+                        text: qsTr("Permissions")
+                    }
+                    Item { Layout.fillWidth: true }
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:702]: how come, it's to place one thing on the right and the other on the left side in the RowLayout.

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml (hunk 1 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:772]: I've replaced the `ColorPicker` with a `StatusPickerButton`, also we need the transitions and when it comes to permissions, we need  also a way to call `pushEditView(properties)` which is a function of the "second page" `PermissionsSettingsPanel` that loads the necessary component.

**Comment by alexjba** [comment]
Not sure the scroll area should be this small and I don't see any designs to confirm it. IMO it's not very practical. I think it would be much better to scroll the entire popup body.

Screenshot 2023-10-31 at 09 34 14

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml [hunk 3]
```diff
@@ -291,8 +297,6 @@ Item {
 
             chatListPopupMenu: ChatContextMenuView {
                 id: chatContextMenuView
-                emojiPopup: root.emojiPopup
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:294]: Not needed?

File: ui/imports/shared/stores/PermissionsStore.qml [hunk 4]
```diff
@@ -14,16 +15,48 @@ QtObject {
...
 
-    readonly property var becomeMemberPermissionsModel: SortFilterProxyModel {
-        id: becomeMemberPermissionsModel
+    readonly property bool viewOnlyCanAddReaction: (qSettings.viewOnlyUserAllowedAddReaction && root.tokenCriteriaMet)
+
+    property var qSettings: Settings {
+        property bool viewOnlyUserAllowedAddReaction: false
+    }
+
+    function setPostRections(checked) {
```
Comment (caybro) [inline ui/imports/shared/stores/PermissionsStore.qml:24]:
```suggestion
    function setPostReactions(checked) {
```

File: ui/imports/shared/stores/PermissionsStore.qml [hunk 5]
```diff
@@ -14,16 +15,48 @@ QtObject {
...
+    }
+
+    readonly property bool tokenCriteriaMet: {
+        for (var i = 0; i < selectedChannelPermissionsModel.count; i++) {
+            var permissionItem = selectedChannelPermissionsModel.itemAt(i);
+            if (permissionItem && permissionItem.tokenCriteriaMet)
+                return true
+        }
+        return false
+    }
```
Comment (caybro) [inline ui/imports/shared/stores/PermissionsStore.qml:44]: How are these 2 properties `viewAndPostCriteriaMet` and `tokenCriteriaMet` different when they contain the same code?

File: ui/app/AppLayouts/Chat/views/ContactsColumnView.qml [hunk 6]
```diff
@@ -141,7 +141,6 @@ Item {
 
                 popupMenu: ChatContextMenuView {
                     id: chatContextMenuView
-                    emojiPopup: root.emojiPopup
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Chat/views/ContactsColumnView.qml:144]: no. the ChatContextMenuView was keeping yet another instance of the CreateChannelPopup so it needed the emojiPopup property to pass it on to it. That instance is now removed so it's not needed anymore.

File: ui/app/AppLayouts/Communities/views/CommunityColumnView.qml (hunk 3 above)
Comment (alexandraB99) [reply ui/app/AppLayouts/Communities/views/CommunityColumnView.qml:294]: yes, see comment above :)

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 7]
```diff
@@ -605,6 +638,113 @@ StatusStackModal {
...
+                    StatusButton {
+                        text: qsTr("Add permission")
+                        onClicked: {
+                            root.currentIndex = 2;
```
[the remark this answers is not in the document]
Comment (alexandraB99) [reply ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml]: done

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 8]
```diff
@@ -169,7 +243,7 @@ StatusStackModal {
             }
             for (let i = 0; i < discordPages.length; i++) {
                 stackItems.push(discordPages[i])
-            }
+            }fc
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:246]: This doesn't look good :)

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 9]
```diff
@@ -88,63 +154,51 @@ StatusStackModal {
...
-                                              emoji,
-                                              colorDialog.color.toString().toUpperCase(),
-                                              root.categoryId)
-                }
-                // TODO Open the channel once we have designs for it
-                root.close()
-            }
         }
     }
+    //TODO
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:193]: What's the `TODO` here?

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 10]
```diff
@@ -205,154 +277,160 @@ StatusStackModal {
...
+                        StatusBaseText {
+                            Layout.topMargin: 8
+                            Layout.alignment: Qt.AlignHCenter
+                            horizontalAlignment: Qt.AlignHCenter
+                            text: qsTr("Export the Discord channel’s chat history data using %1").arg("<a href='https://github.com/Tyrrrz/DiscordChatExporter/releases/tag/2.40.4'>DiscordChatExporter</a>")
+                            onLinkActivated: Global.openLink(link)
+                            HoverHandler {
+                                id: handler1
+                            }
+                            MouseArea {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:350]: No need for the `MouseArea`, set the `cursorShape` on the `HoverHandler` above

File: ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml [hunk 11]
```diff
@@ -205,154 +277,160 @@ StatusStackModal {
...
-                            cursorShape: handler2.hovered && parent.hoveredLink ? Qt.PointingHandCursor : Qt.ArrowCursor
+                        StatusBaseText {
+                            Layout.alignment: Qt.AlignHCenter
+                            horizontalAlignment: Qt.AlignHCenter
+                            text: qsTr("Refer to this <a href='https://github.com/Tyrrrz/DiscordChatExporter/blob/master/.docs/Readme.md'>documentation</a> if you have any queries")
+                            onLinkActivated: Global.openLink(link)
+                            HoverHandler {
+                                id: handler2
+                            }
+                            MouseArea {
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/CreateChannelPopup.qml:364]: Same here, remove `MouseArea`

**Review by caybro — COMMENTED** [review]
Just some code review comments inline, will give it a test locally :)

**Comment by alexjba** [comment]
@alexandraB99 @micieslak @noeliaSD @caybro I've updated this PR to use the Writable proxy model and provide live updates of the proxies once the channel is edited.

Known issues:
1. Once the permissions are created/edited the notifications are duplicated.
2. The entire PR assumes the newly created channel will become the active channel. Not necessarily an issue now because this is happening. But there is a limitation here. We would need a backend change to provide the newly created channel ID when calling the create channel API.
3. The Storybook page doesn't work for the Create new channel flow because the an empty SortFilterProxyModel + ListModel won't provide the proper role models.


I'll create new issues for all these to handle them separately.

**Review by alexandraB99 — COMMENTED** [review]
+1 :) @alexjba could you please check on tests

**Comment by alexjba** [comment]
> +1 :) @alexjba could you please check on tests

Seems to be some community reply not working properly. Let's see if a retry will fix it

**Comment by alexjba** [comment]
Added a new task to fix the multiple notifications being spawned when creating/editing permissions
https://github.com/status-im/status-desktop/issues/13369

**Comment by alexjba** [comment]
@caybro @noeliaSD @micieslak I've updated this PR to break everything into small commits that can be reviewed independently. Please have a look. The related task has been targeted for the RC.

**Review by micieslak — COMMENTED** [review]
Adding permission generates a lot of warnings:
```
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
```

The mechanism preventing creation of identical permission doesn't work in a dialog (but works in settings):
Screenshot from 2024-02-06 17-38-01


The are duplicated notifications, there is also deletion notification when creating one:
Screenshot from 2024-02-06 17-39-11

**Comment by alexjba** [comment]
> Adding permission generates a lot of warnings:
> 
> ```
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> ```
> 
> The mechanism preventing creation of identical permission doesn't work in a dialog (but works in settings): Screenshot from 2024-02-06 17-38-01
> 
> The are duplicated notifications, there is also deletion notification when creating one: Screenshot from 2024-02-06 17-39-11

Thanks @micieslak! I'll have a look on the warnings and the duplicated permission (I've previously tested this and it was working fine).

The duplicated notifications will be covered in a separate task: https://github.com/status-im/status-desktop/issues/13369. I've briefly investigated this assuming we're sending too many requests to the backend, but I've seen the duplicated notifications being triggered even for a single permission update.

**Comment by alexjba** [comment]
> Adding permission generates a lot of warnings:
> 
> ```
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99:13: Unable to assign [undefined] to bool (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:99, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:87, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82:17: Unable to assign [undefined] to QAbstractItemModel* (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:82, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> Warning: qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88:13: Unable to assign [undefined] to int (qrc:/app/AppLayouts/Communities/views/PermissionsView.qml:88, )
> ```

@micieslak Seems to be due to the WritableProxyModel. The `append` operation is a 2 steps operation. First the `QAbstractItemModel::insertRows` and then `set` to fill the row with data. On `insertRows` a new, empty row is created. We'll probably need custom `append` and `insert` when there is also data to be filled.

**Comment by alexjba** [comment]
@micieslak I've pushed the fixes for the warnings (https://github.com/status-im/status-desktop/pull/12160/commits/cd220496d58cd65d56a5c130168a221c0ee96208) and for the duplicate permission detection (https://github.com/status-im/status-desktop/pull/12160/commits/a826ae9411b9c8278ab5618e6ad225c189d21311)

The notifications will be handled in a different task.


https://github.com/status-im/status-desktop/assets/47811206/5bf7daef-c1dd-4447-ba2f-5f600730db37

**Review by noeliaSD — COMMENTED** [review]
Some minor code comments/questions added.

During testing:
1. I cannot see any design that contains this label in the main permissions view, I guess this should not be displayed:

Screenshot 2024-02-07 at 12 14 18

2. I cannot see this option in the app (do I need a special config?):

Screenshot 2024-02-07 at 12 19 27

3. When pressing X a `Save changes` confirmation popup should be [displayed](https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44525%3A270337&mode=design&t=SuPrJJYfAlpCT5oN-1). It's not happening in the app:

https://github.com/status-im/status-desktop/assets/97019400/02a25d1c-09f5-424a-9fcf-ee87a9560d80

4. Minor: It seems there's too much space here compared with the design:

Screenshot 2024-02-07 at 12 28 16

5. Comparing this text color with design, they doesn't seem the same:
App:
Screenshot 2024-02-07 at 12 29 47

Design:
Screenshot 2024-02-07 at 12 29 58

Related to this [text](https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2975%3A489237&mode=design&t=SuPrJJYfAlpCT5oN-1).. I'm wondering if we should use this "no elements" box we have in other places instead of just a text @benjthayer, to have more consistency? I mean like [these boxes](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=17%3A109387&mode=design&t=Aa042YWdWrlh3NGK-1)

6. Not sure if this is one of the known issues but when I create a new channel permission, 3 toasts are displayed (created, deleted and created):

https://github.com/status-im/status-desktop/assets/97019400/d69aa076-e7a5-4b97-bf17-c231c89c7866

File: ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml [hunk 12]
```diff
@@ -0,0 +1,343 @@
...
+        permission.id = Utils.uuid()
+        permission.key = Utils.uuid()
+        permission.holdingsListModel = d.newHoldingsModel(StatusQUtils.ModelUtils.modelToArray(permission.holdingsListModel))
+        permission.channelsListModel = d.newChannelsModel(StatusQUtils.ModelUtils.modelToArray(permission.channelsListModel))
+        channelPermissionsModel.append(permission)
+    }
+
+    // Function removing a permission by index
+    function removePermission(index) {
+        channelPermissionsModel.remove(index, 1);
```
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml:77]: not really..But we can return a bool value saying if the permission has been removed or not. But there's nothing we can do in this case. It's probably a bug somewhere

File: ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml [hunk 13]
```diff
@@ -0,0 +1,343 @@
...
+    }
+
+    // Function duplicating a permission. The new permission will be have a different id and key
+    function duplicatePermission(index) {
+        const permission = channelPermissionsModel.get(index)
+
+        if (!permission)
+            return
+
+        permission.id = Utils.uuid()
```
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml]: I've removed the duplicate function as it's not used anymore. The duplication is handled in the edit channel view

File: ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml [hunk 14]
```diff
@@ -0,0 +1,343 @@
...
+    function appendPermission(holdings, channels, permissionType, isPrivate) {
+        d.appendPermission(holdings, channels, permissionType, isPrivate)
+    }
+
+    // Function editing a permission based on the given arguments
+    function editPermission(key, permissionType, holdings, channels, isPrivate) {
+        d.editPermission(key, permissionType, holdings, channels, isPrivate)
+    }
+
+    // Function duplicating a permission. The new permission will be have a different id and key
```
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/app/AppLayouts/Communities/models/ChannelPermissionsModelEditor.qml]: removed

**Comment by alexjba** [comment]
> Some minor code comments/questions added.
> 
> During testing:
> 
> 1. I cannot see any design that contains this label in the main permissions view, I guess this should not be displayed:
> 
> Screenshot 2024-02-07 at 12 14 18

That's true! Fixed!


> 2. I cannot see this option in the app (do I need a special config?):
> 
> Screenshot 2024-02-07 at 12 19 27

Will be handled in another task. Needs backend https://github.com/status-im/status-desktop/issues/13292; https://github.com/status-im/status-desktop/issues/13291

> 3. When pressing X a `Save changes` confirmation popup should be [displayed](https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=44525%3A270337&mode=design&t=SuPrJJYfAlpCT5oN-1). It's not happening in the app:
> 
>  Screen.Recording.2024-02-07.at.12.23.24.mov 

I'll add a task for this. Wasn't in the scope of this PR and it will probably need some refactoring to do it properly. As I see it we'd need a way to prevent the popup close event from being processed until the user accepts the change. 

> 4. Minor: It seems there's too much space here compared with the design:
> 
> Screenshot 2024-02-07 at 12 28 16

Fixed!

> 5. Comparing this text color with design, they doesn't seem the same:
>    App:
>    Screenshot 2024-02-07 at 12 29 47
> 
> Design: Screenshot 2024-02-07 at 12 29 58
> 
> Related to this [text](https://www.figma.com/file/qHfFm7C9LwtXpfdbxssCK3/Kuba%E2%8E%9CDesktop---Communities?type=design&node-id=2975%3A489237&mode=design&t=SuPrJJYfAlpCT5oN-1).. I'm wondering if we should use this "no elements" box we have in other places instead of just a text @benjthayer, to have more consistency? I mean like [these boxes](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=17%3A109387&mode=design&t=Aa042YWdWrlh3NGK-1)

Fixed

> 6. Not sure if this is one of the known issues but when I create a new channel permission, 3 toasts are displayed (created, deleted and created):
> 
>  Screen.Recording.2024-02-07.at.12.35.26.mov

Yes, known issue to be handled in a different task: https://github.com/status-im/status-desktop/issues/13369


@noeliaSD Please have another look!

**Review by noeliaSD — APPROVED** [review]
Thanks for the fixes!

**Review by caybro — APPROVED** [review]
